usr/local/www/diag_logs.php

```diff
@@ -67,26 +73,28 @@
 include("head.inc");
 
 $tab_array = array();
-$tab_array[] = array(gettext("System"), true, "diag_logs.php");
+$tab_array[] = array(gettext("System"), ($logfile == 'system'), "diag_logs.php");
 $tab_array[] = array(gettext("Firewall"), false, "diag_logs_filter.php");
-$tab_array[] = array(gettext("DHCP"), false, "diag_logs_dhcp.php");
-$tab_array[] = array(gettext("Portal Auth"), false, "diag_logs_auth.php");
-$tab_array[] = array(gettext("IPsec"), false, "diag_logs_ipsec.php");
-$tab_array[] = array(gettext("PPP"), false, "diag_logs_ppp.php");
+$tab_array[] = array(gettext("DHCP"), ($logfile == 'dhcpd'), "diag_logs.php?logfile=dhcpd");
+$tab_array[] = array(gettext("Portal Auth"), ($logfile == 'portalauth'), "diag_logs.php?logfile=portalauth");
+$tab_array[] = array(gettext("IPsec"), ($logfile == 'ipsec'), "diag_logs.php?logfile=ipsec");
+$tab_array[] = array(gettext("PPP"), ($logfile == 'ppp'), "diag_logs.php?logfile=ppp");
 $tab_array[] = array(gettext("VPN"), false, "diag_logs_vpn.php");
-$tab_array[] = array(gettext("Load Balancer"), false, "diag_logs_relayd.php");
-$tab_array[] = array(gettext("OpenVPN"), false, "diag_logs_openvpn.php");
-$tab_array[] = array(gettext("NTP"), false, "diag_logs_ntpd.php");
+$tab_array[] = array(gettext("Load Balancer"), ($logfile == 'relayd'), "diag_logs.php?logfile=relayd");
+$tab_array[] = array(gettext("OpenVPN"), ($logfile == 'openvpn'), "diag_logs.php?logfile=openvpn");
+$tab_array[] = array(gettext("NTP"), ($logfile == 'ntpd'), "diag_logs.php?logfile=ntpd");
 $tab_array[] = array(gettext("Settings"), false, "diag_logs_settings.php");
 display_top_tabs($tab_array);
 
 $tab_array = array();
-$tab_array[] = array(gettext("General"), true, "/diag_logs.php");
-$tab_array[] = array(gettext("Gateways"), false, "/diag_logs_gateways.php");
-$tab_array[] = array(gettext("Routing"), false, "/diag_logs_routing.php");
-$tab_array[] = array(gettext("Resolver"), false, "/diag_logs_resolver.php");
-$tab_array[] = array(gettext("Wireless"), false, "/diag_logs_wireless.php");
-display_top_tabs($tab_array, false, 'nav nav-tabs');
+if(($logfile == 'system') || ($logfile == 'gateways') || ($logfile == 'routing') || ($logfile == 'resolver') || ($logfile == 'wireless')) {
```

Comment:
Thanks. I’ll do that.


> On May 4, 2015, at 10:12 AM, SjonHortensius <notifications@github.com> wrote:
> 
> In usr/local/www/diag_logs.php <https://github.com/SjonHortensius/pfsense/pull/160#discussion_r29587103>:
> 
> >  $tab_array[] = array(gettext("Settings"), false, "diag_logs_settings.php");
> >  display_top_tabs($tab_array);
> >  
> >  $tab_array = array();
> > -$tab_array[] = array(gettext("General"), true, "/diag_logs.php");
> > -$tab_array[] = array(gettext("Gateways"), false, "/diag_logs_gateways.php");
> > -$tab_array[] = array(gettext("Routing"), false, "/diag_logs_routing.php");
> > -$tab_array[] = array(gettext("Resolver"), false, "/diag_logs_resolver.php");
> > -$tab_array[] = array(gettext("Wireless"), false, "/diag_logs_wireless.php");
> > -display_top_tabs($tab_array, false, 'nav nav-tabs');
> > +if(($logfile == 'system') || ($logfile == 'gateways') || ($logfile == 'routing') || ($logfile == 'resolver') || ($logfile == 'wireless')) {
> You can rewrite this to something more readable, use:
> 
> if (in_array($logfile, array('system', 'gateways', 'routing' ...)))
> —
> Reply to this email directly or view it on GitHub <https://github.com/SjonHortensius/pfsense/pull/160/files#r29587103>.
> 

Steve
--
Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com <mailto:sbeaver@netgate.com> | www.netgate.com <http://www.netgate.com/>